utils/parametric-components.stanza

```diff
@@ -0,0 +1,537 @@
+  ; metadata.digi-key-part-number
+  ; metadata.features
+  ; metadata.image
+  ; metadata.number-of-terminations
+  ; metadata.packaging
+  ; metadata.series
+  ; metadata.supplier-device-package
+  ; minimum_quantity
+  ; mounting
+  ; mpn
+  ; price
+  ; rated-power
+  ; rated-temperature.max
+  ; rated-temperature.min
+  ; resistance
+  ; stock
+  ; tcr.neg
+  ; tcr.pos
+  ; tolerance
+  ; tolerance.max
+  ; tolerance.min
+  ; trust
+  ; type
+  ; update_date
+  ; vendor_part_numbers.<VENDOR_STRING>
+  ; x
+  ; y
+  ; z
+
+  ; re-enable when spice is available in jitx-client Docker image for CI
+  ; spice:
+  ;   "[R] <p[1]> <p[2]> <property(self.resistance)>"
+
+public defn parametric-resistor (res:Double|Toleranced) :
+  parametric-resistor(["resistance" => res])
+
+public defn parametric-resistor (res:Double|Toleranced, tol:Double, pkg:String) :
+  parametric-resistor(["resistance" => res "tolerance" => tol "rated-power" => 0.0625 "case" => pkg])
+
+public defn parametric-resistor (res:Double|Toleranced, tol:Double, pwr:Double) :
+  parametric-resistor(["resistance" => res "tolerance" => tol "rated-power" => pwr "case" => "0402"])
+
+public defn parametric-resistor (res:Double|Toleranced, pkg:String) :
+  parametric-resistor(["resistance" => res "tolerance" => 2.0 "rated-power" => 0.0625 "case" => pkg])
+
+public defn parametric-resistor (res:Double|Toleranced, tol:Double) -> Instantiable :
+  parametric-resistor(["resistance" => res "tolerance" => tol])
+
+
+; ========================================================
+; ========== parametric part helper functions ============
+; ========================================================
+defn c-r (v:Double e:Int) :
+  v * pow(10.0, to-double(e))
+
+
+  ; Returns a value within the range of v in the tolerance range specified by tol
+public defn between-std-val (v:Toleranced, tol:Double) -> Double :
+  val expon-min = floor(log10(min-value(v)))
+  val expon-max = floor(log10(max-value(v)))    
+  var match-exp = 0.0
+  var match-val = -1.0
+  var match-diff = 1.0e6
+  ; println("Need to find exp range (%_ %_) for range (%_)" % [expon-min, expon-max, v])
+  for exp in to-int(expon-min) to to-int(expon-max + 1.0) do :
+    for std-val in union-std-vals(tol) as List<Double> do :
+      ; println("in-range?(%_, %_)" % [v, c-r(std-val, exp)])
+      if in-range?(v, c-r(std-val, exp)) :
+        val diff = abs(c-r(std-val, exp) - center-value(v))
+        if diff < match-diff :
+          ; std-val component (at the appropriate exp range) is closest to center()
+          match-diff = diff
+          match-val = std-val
+          match-exp = to-double(exp)
+  match-val * pow(10.0, match-exp)
+
+public defn find-std-val (v:Toleranced) -> Double:
+  var done = 0
+  var ans = 0.0
+  val my-val = 
+    label<Double> return :
+      for tol in std-tols do :
+        ans = between-std-val(v, tol)
+        ; println("Debug tol(%_) ans(%_)" % [tol, ans])
+        return(ans) when ans > 0.0
```

Comment:
Oh, are you using `-1` as default in the function above as a hack to discard it here? You can just return `Maybe<Double>`. Then:
```
val my-val = value! $ first?(between-std-val{v, _}, std-told)
```